File: web/frontend/src/api/credits.ts
// web/frontend/src/api/credits.ts
import client from "./client";

// --- Types ---

export interface PackageResponse {
  id: number;
  name: string;
  price_cents: number;
  credits: number;
  bonus_credits: number;
}

export interface BalanceResponse {
  balance: number;
  total_recharged: number;
  total_consumed: number;
}

export interface RechargeRequest {
  package_id: number;
  pay_method: "pc_web" | "h5";
}

export interface RechargeResponse {
  order_id: number;
  pay_url: string;
}

export interface OrderResponse {
  id: number;
  out_trade_no: string;
  amount_cents: number;
  credits_granted: number;
  status: string;
  pay_method: string;
  created_at: string;
  paid_at: string | null;
}

export interface TransactionResponse {
  id: number;
  type: string;
  amount: number;
  balance_after: number;
  remark: string | null;
  created_at: string;
}

export interface TransactionListResponse {
  items: TransactionResponse[];
  total: number;
  page: number;
  size: number;
}

// --- API Functions ---

export async function getPackages(): Promise<PackageResponse[]> {
  const resp = await client.get<PackageResponse[]>("/credits/packages");
  return resp.data;
}

export async function getBalance(): Promise<BalanceResponse> {
  const resp = await client.get<BalanceResponse>("/credits/balance");
  return resp.data;
}

export async function createRecharge(
  req: RechargeRequest
): Promise<RechargeResponse> {
  const resp = await client.post<RechargeResponse>("/credits/recharge", req);
  return resp.data;
}

export async function getOrder(orderId: number): Promise<OrderResponse> {
  const resp = await client.get<OrderResponse>(`/credits/orders/${orderId}`);
  return resp.data;
}

export async function getTransactions(params?: {
  type?: string;
  page?: number;
  size?: number;
}): Promise<TransactionListResponse> {
  const resp = await client.get<TransactionListResponse>(
    "/credits/transactions",
    { params }
  );
  return resp.data;
}

File: web/src/aigc_web/schemas/credits.py
# web/src/aigc_web/schemas/credits.py
"""积分相关的请求/响应模型。"""

from datetime import datetime
from typing import Literal

from pydantic import BaseModel


# --- 套餐 ---

class PackageResponse(BaseModel):
    id: int
    name: str
    price_cents: int
    credits: int
    bonus_credits: int

    model_config = {"from_attributes": True}


# --- 充值 ---

class RechargeRequest(BaseModel):
    package_id: int
    pay_method: Literal["pc_web", "h5"]


class RechargeResponse(BaseModel):
    order_id: int
    pay_url: str


# --- 订单 ---

class OrderResponse(BaseModel):
    id: int
    out_trade_no: str
    amount_cents: int
    credits_granted: int
    status: str
    pay_method: str
    created_at: datetime
    paid_at: datetime | None


# --- 流水 ---

class TransactionResponse(BaseModel):
    id: int
    trade_no: str
    type: str
    amount: int
    balance_after: int
    remark: str | None
    created_at: datetime

    model_config = {"from_attributes": True}


class TransactionListResponse(BaseModel):
    items: list[TransactionResponse]
    total: int
    page: int
    size: int


# --- 余额 ---

class BalanceResponse(BaseModel):
    balance: int
    total_recharged: int
    total_consumed: int
